modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/AbstractEventEndpoint.java

```diff
@@ -1172,6 +1172,29 @@ public Response createEventCommentReply(@PathParam("eventId") String eventId, @P
     }
   }
 
+  /**
+   * Removes emtpy series titles from the collection of the isPartOf Field
+   * @param ml the list to modify
+   */
+  private void removeSeriesWithNullTitlesFromFieldCollection(MetadataList ml) {
+    // get Series MetadataField from MetadataList
+    MetadataField seriesField = Optional.ofNullable(ml.getMetadataList().get("dublincore/episode"))
+            .flatMap(titledMetadataCollection -> Optional.ofNullable(titledMetadataCollection.getCollection()))
+            .flatMap(dcMetadataCollection -> Optional.ofNullable(dcMetadataCollection.getOutputFields()))
+            .flatMap(metadataFields -> Optional.ofNullable(metadataFields.get("isPartOf")))
+            .orElse(null);
+    if (seriesField == null || seriesField.getCollection() == null) {
+      return;
+    }
+
+    // Remove null keys
+    Map<String, String> seriesCollection = seriesField.getCollection();
+    while (seriesCollection.remove(null) != null) { }
```

Comment:
This is just a confusing way of just doing:
```suggestion
    seriesCollection.remove(null);
```
Map keys have to be unique, or don't they? That means there can only be one element with key `null` in the `Map`. That means you only ever call map twice in this loop and the last time it will never do anything.

Reply:
Yes, of course.